docs/SCHEMA.md

```diff
@@ -209,3 +211,15 @@ In general, feel free to add extra entities that tracks the internal state of th
 Make sure these changes are **strictly** additive. If you need to **modify** existing entities, please let me know and we can go through it together.
 
 When adding entities, make sure you document them with comments in your schema. User-facing comments should be done in double quotes and developer-facing comments should be done with hash tags.
+
+## Immutable Field Assumptions
+
+There are some fields that we (and subgraph consumers) are going to expect to be immutable. There are technical limitations as to why we cannot force this on a subgraph level, but we want to be very cautious when we change these fields in `prod` subgraphs especially.
+
+Scenario: We expect the `market.id` field to not change on deployments, and a DB ingesting our subgraph will expect this to be the case. But then we change `market.id` and the DB thinks there is a new `market` when in actuality, we just updated the identifier.
+
+In general you should not change the following fields across deployments before consulting with the downstream users:
```

Comment:
I think the actual fields that are mutable/immutable can be fairly subjective.
Do you think we should add a label or additional comment on the subgraph schema for each field that's immutable?

Reply:
Yeah we could do that, is there anything you think I missed in the list?

Reply:
It seems like overkill to say that deposit.id should be immutable (they are now immutable entities, but it could still have diff IDs across versions, technically)

Reply:
Yep agreed, therefore I think it's better to define and note it down on a per-field-per-entity level, rather than blanket declare that ids are all immutable.

Reply:
I have put this on hold for a while, but it doesn't feel right to add this to every immutable field in the schemas imo.

Feels unnecessary since we have notes here. Really the IDs are the most important field to remain immutable for our downstream ingestion.